tests/js/spec/views/organizationDiscover/utils.spec.jsx

```diff
@@ -86,26 +87,64 @@ const sampleQueryData = [
 
 const fields = ['platform', 'exception_stacks.type'];
 
+function getLabel(value) {
+  if (typeof value === 'object') {
+    try {
+      value = JSON.stringify(value);
+    } catch (err) {
+      // eslint-disable-next-line no-console
+      console.error(err);
+    }
+  }
+
+  return value;
+}
+
 function getDataForChart(queryData, groupbyFields) {
+  // const {aggregations} = this.props.query;
+  // const aggregate = aggregations[0][2];
+  const aggregate = 'count';
+
+  const rawDates = [...new Set(queryData.map(entry => entry.time))];
+
+  const formattedDates = rawDates.map(time => moment(time * 1000).format('MM-DD'));
+
   const output = {};
   queryData.forEach(data => {
-    const key = groupbyFields.map(field => data[field]).join(',');
-    if (key in output) {
-      output[key].count.push(data.count);
-    } else {
-      output[key] = {count: [data.count]};
+    const key = groupbyFields.map(field => getLabel(data[field])).join(',');
+    const val = getLabel(data[aggregate]);
+    const idx = rawDates.indexOf(data.time);
+
+    if (!(key in output)) {
+      output[key] = {};
     }
+    output[key][idx] = val;
   });
 
-  return output;
+  // If there is no data for that series in that time period, fill it with null
+  const data = {};
+  Object.entries(output).forEach(([line, val]) => {
+    data[line] = new Array(rawDates.length).fill(null);
```

Comment:
Why `null` instead of `0`?

Reply:
@lynnagara

Reply:
This is to distinguish the absence of a value from an actual zero value for that point as they aren't the same. For example, a limit parameter set lower than the length of the result set might cause incomplete results to be displayed on the chart even though there are actual events for those periods.